File: summary/ruby/16_eratosthenes.rb
def sieve_of_eratosthenes(max)
  numbers = (2..max).to_a
  primes = []

  while numbers.any?
    prime = numbers.shift
    primes << prime
    numbers.reject! { |num| num % prime == 0 }
  end

  primes
end

# Example usage:
max_number = 30
primes = sieve_of_eratosthenes(max_number)
puts "Primes up to #{max_number}: #{primes.join(', ')}"

File: summary/ts/16_eratosthenes.ts
function sieveOfEratosthenes(max: number): number[] {
    let numbers: number[] = Array.from({ length: max - 1 }, (_, i) => i + 2);
    let primes: number[] = [];

    while (numbers.length > 0) {
        const prime = numbers.shift()!;
        primes.push(prime);
        numbers = numbers.filter(num => num % prime !== 0);
    }

    return primes;
}

// Example usage:
const maxNumber: number = 30;
const primes: number[] = sieveOfEratosthenes(maxNumber);
console.log(`Primes up to ${maxNumber}: ${primes.join(', ')}`);
